nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
What's the intention behind this? I don't see it being obtainable from the API - caching is an implementation detail, isn't it?